pkg/storage/edv/documentprocessor/documentprocessor.go

```diff
@@ -7,12 +7,91 @@ SPDX-License-Identifier: Apache-2.0
 package documentprocessor
 
 import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/jose"
 	"github.com/hyperledger/aries-framework-go/pkg/storage/edv"
 )
 
+const (
+	failMarshalStructuredDocument   = "failed to marshal structured document into bytes: %w"
+	failEncryptStructuredDocument   = "failed to encrypt structured document into JWE form: %w"
+	failJWESerialize                = "failed to serialize JWE: %w"
+	failDeserializeJWE              = "failed to deserialize JWE: %w"
+	failDecryptJWE                  = "failed to decrypt JWE: %w"
+	failUnmarshalStructuredDocument = "failed to unmarshal structured document: %w"
+)
+
+type marshalFunc func(interface{}) ([]byte, error)
+
 // DocumentProcessor represents a type that can encrypt and decrypt between
 // Structured Documents and Encrypted Documents.
 type DocumentProcessor interface {
```

Comment:
Where is this interface consumed?

The DocumentProcessor shouldn't be coupled to EDV so you should consider another package for it.

Reply:
This is used by the (EDV) `FormatProvider` for doing encryption/decryption before calling the underlying store. This `DocumentProcessor` and the `FormatProvider` together are what allows me to decouple the EDV REST API from the underlying store.

It's specifically for the EDV Structured Documents and Encrypted Document models as defined in the spec. It's not for general purpose encryption/decryption. Can you clarify what you mean by decoupling it from EDV?

Reply:
You shouldn’t normally be exporting both an interface and the concrete implementation from the same package.

Reply:
Fixed.